vendor_pricelist_category/models/product_supplierinfo_category.py

```diff
@@ -70,13 +71,26 @@ class ProductSupplierinfoCategory(models.Model):
         recursive=True,
     )
 
-    _sql_constraints = [
-        (
-            "code_company_unique",
-            "unique(code, company_id)",
-            _("The code is already in use."),
-        )
-    ]
+    @api.constrains("code", "company_id", "parent_id")
+    def _check_code(self):
+        for category in self.filtered("code"):
+            categories = (
+                self.env["product.supplierinfo.category"]
+                .search(
+                    [
+                        ("parent_id", "=", category.parent_id.id),
+                        ("company_id", "=", category.company_id.id),
+                    ]
+                )
+                .filtered(lambda a, category=category: a.code == category.code)
```

Comment:
Why is this with a filtered and not added to the search?

Reply:
Only at the same level